aws/resource_aws_iot_thing_group.go

```diff
@@ -0,0 +1,274 @@
+}
+
+func resourceAwsIotThingGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iotconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	input := &iot.DescribeThingGroupInput{
+		ThingGroupName: aws.String(d.Id()),
+	}
+	log.Printf("[DEBUG] Reading IoT Thing Group: %s", input)
+	out, err := conn.DescribeThingGroup(input)
+
+	if err != nil {
+		if isAWSErr(err, iot.ErrCodeResourceNotFoundException, "") {
+			log.Printf("[WARN] IoT Thing Group %q not found, removing from state", d.Id())
+			d.SetId("")
+		}
+		return err
+	}
+	log.Printf("[DEBUG] Received IoT Thing Group: %s", out)
+
+	d.Set("arn", out.ThingGroupArn)
+	d.Set("name", out.ThingGroupName)
+
+	if err := d.Set("metadata", flattenIotThingGroupMetadata(out.ThingGroupMetadata)); err != nil {
+		return fmt.Errorf("error setting metadata: %s", err)
+	}
+	if err := d.Set("properties", flattenIotThingGroupProperties(out.ThingGroupProperties)); err != nil {
+		return fmt.Errorf("error setting properties: %s", err)
+	}
+	d.Set("version", out.Version)
+
+	tags, err := keyvaluetags.IotListTags(conn, *out.ThingGroupArn)
+	if err != nil {
+		return fmt.Errorf("error listing tags for Iot Thing Group (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsIotThingGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iotconn
+
+	input := &iot.UpdateThingGroupInput{
+		ThingGroupName: aws.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("properties"); ok {
+		input.ThingGroupProperties = expandIotThingsGroupProperties(v.([]interface{}))
+	}
+
+	if d.HasChange("tags") {
+		oldTags, newTags := d.GetChange("tags")
+
+		if v, ok := d.GetOk("arn"); ok {
+			if err := keyvaluetags.IotUpdateTags(conn, v.(string), oldTags, newTags); err != nil {
+				return fmt.Errorf("error updating Iot Thing Group (%s) tags: %s", d.Id(), err)
+			}
+		}
+	}
+
+	_, err := conn.UpdateThingGroup(input)
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsIotThingGroupRead(d, meta)
+}
+
+func resourceAwsIotThingGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iotconn
+
+	input := &iot.DeleteThingGroupInput{
+		ThingGroupName: aws.String(d.Id()),
+	}
+	log.Printf("[DEBUG] Deleting IoT Thing Group: %s", input)
+
+	_, err := conn.DeleteThingGroup(input)
+	if err != nil {
+		if isAWSErr(err, iot.ErrCodeResourceNotFoundException, "") {
+			return nil
+		}
+		return err
+	}
+
+	return nil
+}
+
+func expandIotThingsGroupProperties(l []interface{}) *iot.ThingGroupProperties {
+	m := l[0].(map[string]interface{})
+
+	thingGroupProperties := &iot.ThingGroupProperties{}
+
+	if v, ok := m["attributes"]; ok {
+		thingGroupProperties.AttributePayload = &iot.AttributePayload{
+			Attributes: stringMapToPointers(v.(map[string]interface{})),
```

Comment:
As per git commit:
commit cd020182d2332d17441ccb602d6820c7eca6e54f
Author: Angie Pinilla <angelinepinilla@gmail.com>
Date:   Wed Apr 14 14:03:50 2021 -0400

    migrate stringMapToPointers usage to expandStringMap


I think you will need to change this to use:  expandStringMap